lib/ad_localize/mappers/android_translation_mapper.rb

```diff
@@ -9,9 +9,17 @@ def sanitize_value(value:)
         processedValue = processedValue.gsub(/(?<!\\)\"/, '&#34;') # match " unless there is a \ before
         processedValue = processedValue.gsub(">", '&gt;')
         processedValue = processedValue.gsub("<", '&lt;')
+        hasFormatting = processedValue.match(/(%(\d+\$)?@)/)
         processedValue = processedValue.gsub(/(%(\d+\$)?@)/, '%\2s') # should match values like %1$s and %s
+        hasFormatting = hasFormatting || processedValue.match(/(%((\d+\$)?(\d+)?)i)/)
         processedValue = processedValue.gsub(/(%((\d+\$)?(\d+)?)i)/, '%\2d') # should match values like %i, %3$i, %01i, %1$02i
-        processedValue = processedValue.gsub(/%(?!((\d+\$)?(s|(\d+)?d)))/, '&#37;') # negative lookahead: identifies when user really wants to display a %
+        # On Android, '%' must be escaped with a second '%' if and only if the string has at least one formatting pattern. In this specific case, 
+        # a Java formatting method will be used which interprets every non escaped '%' as the start of a formatting pattern.
```

Comment:
Does the Java formatting method work with a string or a file scope ?
I mean will all occurrences of % be interpreted if the file contains one formatting pattern  ?

Reply:
The Java formatter works with a string, not with a file. It is called when we call `context.getString(R.string.{RESOURCE_ID}, {NOT-EMPTY-PARAMETERS})` and will only be applied this time for the localizable `RESOURCE_ID`.